dss/storage/validator.py

```diff
@@ -32,34 +100,28 @@ def remove_json_fields(json_data: dict, path: List[str], fields: List[str]):
         current.pop(field)
 
 
-DSS_Draft4Validator = validators.create(
-    meta_schema=_utils.load_schema("draft4"),
-    validators={
-        u"$ref": _validators.ref,
-        u"additionalProperties": _validators.additionalProperties,
-        u"properties": _validators.properties_draft4,
-        u"required": _validators.required_draft4,
-    },
-    version="draft4",
-)
-
-resolver = validators.RefResolver(referrer='', base_uri='')
-
-
 def scrub_index_data(index_data: dict, bundle_id: str, logger: logging.Logger) -> list:
+    handler = S3UrlCache(logger)
+    resolver = validators.RefResolver(referrer='',
+                                      base_uri='',
+                                      handlers={'http': handler.request_json,
+                                                'https': handler.request_json}
+                                      )
     extra_fields = []
     extra_documents = []
     for document in index_data.keys():
         core = index_data[document].get('core', None)
         if core is not None:
             try:
-                schema = read_json_url(core['schema_url'])
-            except URLError as ex:
+                schema = handler.request_json(core['schema_url'])
+            except requests.ConnectionError as ex:
                 extra_documents.append(document)
-                logger.warning("%s", f"Unable to retrieve schema from url {core['schema_url']} due to exception: {ex}.")
+                logger.warning("%s",
+                               f"Unable to retrieve schema from url {core['schema_url']} due to exception: {ex}.")
```

Comment:
Nit: Extracting `core['schema_url']` into a variable would guarantee that the error message never gets out of sync with reality.